File: Python/chapter 3/2018.py
n = int(input())
start = 1
end = 1
sum = 1
count = 1

while end != n:
    if(sum > n):
        sum -= start
        start += 1
    elif(sum < n):
        end += 1
        sum += end
    elif(sum == n):
        count +=1
        end += 1
        sum += end

print(count)

File: CPP/chapter3/2018.cpp
#include <iostream>

using namespace std;

int main(){

    int n;
    int start_index = 1;
    int end_index = 1;
    int count = 1;
    int sum = 1;

    cin >> n;

    while(end_index != n){

        if(sum == n){
            count ++;
            end_index ++;
            sum = sum + end_index;
        }

        else if(sum > n){
            sum = sum - start_index;
            start_index ++;
        }
        else{
            end_index++;
            sum  = sum + end_index;
        }

    }

    cout << count << "\n";

}